mesonbuild/dependencies/ui.py

```diff
@@ -493,3 +493,56 @@ def check_wxconfig(self):
                 pass
         WxDependency.wxconfig_found = False
         mlog.log('Found wx-config:', mlog.red('NO'))
+        
+class VulkanDependency(ExternalDependency):
+    def __init__(self, environment, kwargs):
+        super().__init__('vulkan', environment, None, kwargs)
+        
+        if DependencyMethods.SYSTEM in self.methods:
+            try:
+                self.vulkan_sdk = os.environ['VULKAN_SDK']
+                if not os.path.isabs(self.vulkan_sdk):
+                    raise DependencyException('VULKAN_SDK must be an absolute path.')
+                    
+                # TODO: this config might not work on some platforms, fix bugs as reported
+                # we at least have to detect other 64-bit platforms (e.g. armv8)
+                lib_name = 'vulkan'
+                if mesonlib.is_windows():
+                    lib_name = 'vulkan-1'
+                    
+                lib_dir = 'Lib32'
+                if detect_cpu({}) == 'x86_64':
+                    lib_dir = 'Lib'
+                    
+                self.type_name = 'vulkan_sdk'
+                self.is_found = True
+                self.compile_args.append('-I' + os.path.join(self.vulkan_sdk, 'Include'))
+                self.link_args.append('-L' + os.path.join(self.vulkan_sdk, lib_dir))
+                self.link_args.append('-l' + lib_name)
```

Comment:
when you load the whole api at runtime you will probably need another header file (with function pointers and stuff) anyways. The gl dependency does not have an option for it either.